src/main/java/paymelah/model/debt/DebtList.java

```diff
@@ -0,0 +1,116 @@
+package paymelah.model.debt;
+
+import static java.util.Objects.requireNonNull;
+import static paymelah.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import paymelah.model.debt.exceptions.DebtNotFoundException;
+
+/**
+ * Represents a list of a person's {@link Debt} in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class DebtList {
+    private final List<Debt> debts = new ArrayList<>();
+
+    /**
+     * Constructs a {@code DebtList}.
+     * {@code debts} must be present and all debts must be not null.
+     *
+     * @param debts The list of debts owed.
+     */
+    public DebtList(DebtList debts) {
+        requireAllNonNull(debts);
+        this.debts.addAll(debts.debts);
+    }
+
+    /**
+     * Constructs an empty {@code DebtList}.
+     */
+    public DebtList() {}
+
+    /**
+     * Returns true if the list contains an equivalent person as the given argument.
```

Comment:
should this be equivalent debt instead of equivalent person?

Reply:
Fixed.